File: simplypdf/src/main/java/com/wwdablu/soumya/simplypdf/jsonengine/BaseConverter.java
package com.wwdablu.soumya.simplypdf.jsonengine;

import com.wwdablu.soumya.simplypdf.composers.Composer;

import org.json.JSONException;
import org.json.JSONObject;

abstract class BaseConverter {

    protected abstract void generate(Composer composer, JSONObject compose) throws Exception;

    protected String getProperties(JSONObject compose) throws JSONException {
        return compose.has(Node.TYPE_PROPERTIES) ?
                compose.getJSONObject(Node.TYPE_PROPERTIES).toString() : null;
    }
}

File: simplypdf/src/main/java/com/wwdablu/soumya/simplypdf/jsonengine/BaseConverter.kt
package com.wwdablu.soumya.simplypdf.jsonengine

import com.wwdablu.soumya.simplypdf.composers.Composer
import org.json.JSONException
import org.json.JSONObject

internal abstract class BaseConverter {
    @Throws(Exception::class)
    protected abstract fun generate(composer: Composer, compose: JSONObject)

    @Throws(JSONException::class)
    protected fun getProperties(compose: JSONObject): String? {
        return if (compose.has(Node.TYPE_PROPERTIES)) compose.getJSONObject(Node.TYPE_PROPERTIES)
            .toString() else null
    }
}